### PR TITLE
Add real dark theme to modal and popover (issue/2491)

### PR DESCRIPTION
**Pull request by henrijoss**
### Pull Request Checklist

- [x] I have read and accept the [Contributor License Agreement](https://opensource.porsche.com/docs/cla)

#### References

- Preview: https://designsystem.porsche.com/issue/2491

#### Scope

Short description of the pull request scope

#### Resolved Issue

Resolves [#2491](https://github.com/porsche-design-system/porsche-design-system/issues/2491)

### CODE REVIEW

File: packages/components-js/tests/vrt/playwright/specs/__screenshots__/modal-1000-theme-dark-chrome.png
Comment (denyo) [inline packages/components-js/tests/vrt/playwright/specs/__screenshots__/modal-1000-theme-dark-chrome.png:1]: Looks like contrast is to weak? Not sure if modal background or playground background color is the issue but when you compare it to the popover it is noticeable different.

Bildschirmfoto 2023-10-05 um 09 48 47

File: packages/components-js/tests/vrt/playwright/specs/__screenshots__/popover-print-theme-dark-chrome.png
Comment (denyo) [inline packages/components-js/tests/vrt/playwright/specs/__screenshots__/popover-print-theme-dark-chrome.png:1]: Black arrows look like a regression?
Bildschirmfoto 2023-10-05 um 09 51 09

File: packages/components-js/tests/vrt/playwright/specs/__screenshots__/popover-1760-overview-theme-light-chrome.png
Comment (denyo) [inline packages/components-js/tests/vrt/playwright/specs/__screenshots__/popover-1760-overview-theme-light-chrome.png:1]: Where does the 32px height increase come from?

Comment (henrijoss) [reply packages/components-js/tests/vrt/playwright/specs/__screenshots__/popover-1760-overview-theme-light-chrome.png:1]: i added the playground class

File: packages/components/CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,11 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 ### [Unreleased]
 
+#### Added
+
+- `Popover` and `Modal` support dark theme
```
Comment (denyo) [inline packages/components/CHANGELOG.md:19]: and auto :)

File: packages/components-js/tests/vrt/playwright/specs/__screenshots__/popover-print-theme-dark-chrome.png
Comment (henrijoss) [reply packages/components-js/tests/vrt/playwright/specs/__screenshots__/popover-print-theme-dark-chrome.png:1]: Do we already support print view?

File: packages/components-js/tests/vrt/playwright/specs/__screenshots__/modal-1000-theme-dark-chrome.png
Comment (henrijoss) [reply packages/components-js/tests/vrt/playwright/specs/__screenshots__/modal-1000-theme-dark-chrome.png:1]: Popover has a different background color. I'll ask the designers regarding the modal again.

File: packages/components/src/components/modal/modal-styles.ts [hunk 2]
```diff
@@ -219,6 +241,10 @@ export const getComponentCss = (
             borderColor: darkThemeContrastHighColor,
           },
         }),
+        ...prefersColorSchemeDarkMediaQuery(theme, {
+          background: backgroundColorDark,
+          border: `2px solid ${backgroundColorDark}`,
```
Comment (denyo) [inline packages/components/src/components/modal/modal-styles.ts:246]: Could just overwrite `border-color` instead of entire `border` shorthand.

Comment (henrijoss) [reply packages/components/src/components/modal/modal-styles.ts]: it's stronger and should be the correct one

File: packages/components-js/tests/vrt/playwright/specs/custom/popover.vrt.ts [hunk 3]
```diff
@@ -15,7 +15,7 @@ const scenario = async (page: Page, theme: Theme, scheme?: PrefersColorScheme):
   };
 
   const markup =
-    () => `<div style="position: relative; height: 800px; outline: 1rem solid rgba(0, 0, 255, 0.1); outline-offset: -1rem">
+    () => `<div style="position: relative; height: 800px; outline: 1rem solid rgba(0, 0, 255, 0.1); outline-offset: -1rem" class="playground">
```
Comment (stephanschroeter) [inline packages/components-js/tests/vrt/playwright/specs/custom/popover.vrt.ts:18]: I think the playground class isn't necessary in this context?

File: packages/components-js/tests/vrt/playwright/specs/__screenshots__/popover-print-theme-dark-chrome.png
Comment (stephanschroeter) [reply packages/components-js/tests/vrt/playwright/specs/__screenshots__/popover-print-theme-dark-chrome.png:1]: We don't support print view yet, so you can ignore the black arrows for now.

File: packages/components-js/tests/vrt/playwright/specs/__screenshots__/popover-1760-overview-theme-light-chrome.png
Comment (stephanschroeter) [reply packages/components-js/tests/vrt/playwright/specs/__screenshots__/popover-1760-overview-theme-light-chrome.png:1]: I think the playground class isn't necessary in this context.

Comment (henrijoss) [reply packages/components-js/tests/vrt/playwright/specs/__screenshots__/popover-1760-overview-theme-light-chrome.png:1]: the background color doesn't get set without the playground class

File: packages/components-js/tests/vrt/playwright/specs/custom/popover.vrt.ts (hunk 3 above)
Comment (henrijoss) [reply packages/components-js/tests/vrt/playwright/specs/custom/popover.vrt.ts:18]: Necessary for `waitForForcedComponentTheme` to set the theme class

File: packages/components/CHANGELOG.md [hunk 4]
```diff
@@ -14,6 +14,11 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 ### [Unreleased]
 
+#### Added
+
+- `Popover` and `Modal` support dark and auto theme & changed `themeDarkBackgroundShading` color
+  ([#2789](https://github.com/porsche-design-system/porsche-design-system/pull/2789))
```
Comment (stephanschroeter) [inline packages/components/CHANGELOG.md:20]:
```suggestion
- `Popover` and `Modal` support dark and auto theme ([#2789](https://github.com/porsche-design-system/porsche-design-system/pull/2789))
- Styles: changed `themeDarkBackgroundShading` color
  ([#2789](https://github.com/porsche-design-system/porsche-design-system/pull/2789))
```

File: packages/components/CHANGELOG.md [hunk 5]
```diff
@@ -14,6 +14,16 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
+#### Added
+
+- `Popover` and `Modal` support theme dark and auto
+  ([#2789](https://github.com/porsche-design-system/porsche-design-system/pull/2789))
+
+#### Changed
+
+- Styles: changed `themeDarkBackgroundShading` color
+  ([#2789](https://github.com/porsche-design-system/porsche-design-system/pull/2789))
+
```
Comment (stephanschroeter) [inline packages/components/CHANGELOG.md:26]:
```suggestion
- Styles: `themeDarkBackgroundShading` and `pds-theme-dark-background-shading` color
  ([#2789](https://github.com/porsche-design-system/porsche-design-system/pull/2789))
```

**Review by stephanschroeter — APPROVED** [review]

**Pull request merged**